src/tools/illink/src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -27,7 +27,53 @@ internal partial struct HandleCallAction
 		private readonly FlowAnnotations _annotations;
 		private readonly RequireDynamicallyAccessedMembersAction _requireDynamicallyAccessedMembersAction;
 
-		public bool Invoke (MethodProxy calledMethod, MultiValue instanceValue, IReadOnlyList<MultiValue> argumentValues, IntrinsicId intrinsicId, out MultiValue methodReturnValue)
+		public void Invoke (MethodProxy calledMethod, MultiValue instanceValue, IReadOnlyList<MultiValue> argumentValues, IntrinsicId intrinsicId, out MultiValue methodReturnValue)
+		{
+			MultiValue? maybeMethodReturnValue;
+
+			if (!TryHandleIntrinsic (calledMethod, instanceValue, argumentValues, intrinsicId, out maybeMethodReturnValue))
+				HandleSharedIntrinsic (calledMethod, instanceValue, argumentValues, intrinsicId, out maybeMethodReturnValue);
+
+			// If we get here, we handled this as an intrinsic.  As a convenience, if the code above
+			// didn't set the return value (and the method has a return value), we will set it to be an
+			// unknown value with the return type of the method.
```

Comment:
I know this is just moved from existing code, but ideally we'd want the intrinsic handling above to always set this rather than setting a default here, right? If so we should consider filing an issue to track it.